site/src/sphinx/server-decorator.rst

```diff
@@ -217,6 +217,69 @@ Therefore, you need to specify the decorators as extra parameters:
 A good real-world example of :api:`ServiceWithRoutes` is :api:`GrpcService`.
 See :ref:`server-grpc-decorator` for more information.
 
+Decorating a service by path mapping
+------------------------------------
+
+If you want to decorate a :api:`Service` by path mapping or router matching, you can specify a way to select
+the services which are being decorated with ``decoratorUnder(pathPrefix, ...)`` or ``decorator(Route, ...)``.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.common.HttpHeaderNames;
+
+    VipService vipService = ...;
+    MemberService memberService = ...;
+    HtmlService htmlService = ...;
+    JsService jsService = ...;
+
+    ServerBuilder sb = new ServerBuilder();
+    sb.annotatedService("/users/vip", vipService)
+      .annotatedService("/users/members", memberService)
+      // decorate all services under '/users' path
+      .decoratorUnder("/users", (delegate, ctx, req) -> {
+          if (!authenticate(req)) {
+              return HttpResponse.of(HttpStatus.UNAUTHORIZED);
+          }
+          return delegate.serve(ctx, req);
+      })
+      .serviceUnder("/public/html", htmlService)
+      .serviceUnder("/public/js", jsService)
+      // decorate services with Route.builder()...build()
+      .decorator(Route.builder().pathPrefix("/public").build(), (delegate, ctx, req) -> {
+          final HttpResponse response = delegate.serve(ctx, req);
+          ctx.addAdditionalResponseHeader(HttpHeaderNames.CACHE_CONTROL, "public");
+          return response;
+      });
+
+You can also use fluent route builder with ``routeDecorator()`` to match services being decorated.
+
+.. code-block:: java
+
+    ServerBuilder sb = new ServerBuilder();
+    sb.annotatedService("/users/vip", vipService)
+      .annotatedService("/users/members", memberService)
+      // decorate services under '/users' path with fluent route builder
+      .routeDecorator()
+      .pathPrefix("/users")
+      .build((delegate, ctx, req) -> {
+          if (!authenticate(req)) {
+              return HttpResponse.of(HttpStatus.UNAUTHORIZED);
+          }
+          return delegate.serve(ctx, req);
+      })
+      .serviceUnder("/public/html", htmlService)
+      .serviceUnder("/public/js", jsService)
+      .routeDecorator()
```

Comment:
How about splitting the call chain into four parts?

1. annotatedService()s
2. route decorator for /users
3. serviceUnder()s
4. route decorator for /public

i.e.

```java
// Some comment
sb.annotatedService(...)
  .annotatedService(...);

// Some comment
sb.routeDecorator()
  ...
  .build();

// Some comment
sb.serviceUnder(...)
  .serviceUnder(...);

// Some comment
sb.routeDecorator()
  ...
  .build();
```

(Please fix the first example as well.)